### ISSUE TITLE
Localization issues with compiler tests

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn#67743]
At a glance, all these look like legitimate localization issues. Simply disabling the tests without thorough investigation will not be appropriate, I think.

_Originally posted by @AlekseyTs in https://github.com/dotnet/roslyn/issues/67734#issuecomment-1503751153_

See original discussion for further details

**Comment by jaredpar** [comment]
@AlekseyTs helped me get a machine running in Russian this afternoon and I'm pretty sure we understand why our CI is not failing the `NetModuleReference_ModuleSpecifiedAsAssembly` tests. The machine is likely setup such that `CurrentCulture` is `es-ES` and `CurrentUICulture` is `en-EN`. Our method of realizing resource strings goes through `ResourceManager.GetString` and that will use `CurrentUICulture` for formatting. Once I get ahold of one of our CI machines I can verify that is indeed the setup there. If so have a plan for how we can close that gap in our CI. 

Still a bit puzzled by `LambdaReturnInference_04` though.

**Comment by jaredpar** [comment]
Dug a bit more this morning and I finally understand why our .NET Framework Spanish test was not flagging so many of these failures. When we initially dig the [major effort to re-architect](37456ead53c255ab3c8feb6657b3c7e645428d98) the compiler tests such that they could run on .NET Core and .NET Framework we introduced the type `CappedStringWriter`. That was part of unifying how we captured `System.Console.Out` from desktop and core execution. 

The problem is how that class initializes the base `StringWriter` type:

```csharp
        public CappedStringWriter(int expectedLength)
            : base(System.Globalization.CultureInfo.InvariantCulture)
```

For .NET Framework our output is intercepted by roughly calling `Console.SetOut(new CappedStringWriter())`. That means effectively any `System.Console.WriteLine` in our unit tests will format output invariantly. For .NET Core though there is an extra indirection layer between `System.Out` and `CappedStringWriter` which causes normal formatting of values. Thus .NET Framework tests passed and .NET Core tests failed.
